macrosynergy/dataquery/api.py

```diff
@@ -59,24 +62,27 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_value, exc_traceback):
         if exc_type:
-            print(f'Execution {exc_type} with value (exc_value):\n{exc_value}')
+            print(f"Execution {exc_type} with value (exc_value):\n{exc_value}")
 
     def check_connection(self) -> bool:
-        """Check connect (heartbeat) to DataQuery"""
-        js: dict = self.access.get_dq_api_result(url=self.access.base_url + "/services/heartbeat")
-
-        results: dict = js["info"]
-        if int(results["code"]) != 200:
-            print(
-                f"DataQuery response {results['message']:s}"
-                f" with description: {results['description']:s}"
-            )
+        """Check connection (heartbeat) to DataQuery.
+        """
+        endpoint = "/services/heartbeat"
+        js: dict = self.access.get_dq_api_result(url=self.access.base_url + endpoint)
 
-        return int(results["code"]) == 200
+        try:
+            results: dict = js["info"]
+        except KeyError:
+            results: dict = js["errors"][0]
+            print(f"The corresponding code is: {results['code']}.")
+            raise ConnectionError(results["description"])
```

Comment:
To make it more robust lets change it do:
```python
try:
    results: dict = js["info"]
except KeyError:
    raise ConnectionError(f"DataQuery error response: {js}")

return int(results["code"]) == 200, results
```
that way we are independent of the structure of `js` (the response).

Reply:
@Ksteeds